src/main/java/wtf/meier/tariff/interpreter/model/tariff/calculator/DayBasedTariffCalculator.kt

```diff
@@ -1,24 +1,24 @@
 package wtf.meier.tariff.interpreter.model.tariff.calculator
 
+import wtf.meier.tariff.interpreter.model.RentalPeriod
 import wtf.meier.tariff.interpreter.extension.min
 import wtf.meier.tariff.interpreter.model.Receipt
 import wtf.meier.tariff.interpreter.model.extension.toReceipt
 import wtf.meier.tariff.interpreter.model.rate.RateCalculator
 import wtf.meier.tariff.interpreter.model.tariff.DayBasedTariff
-import java.time.Instant
 import java.time.LocalTime.MAX
 import java.time.LocalTime.MIN
 
 class DayBasedTariffCalculator(
     private val rateCalculator: RateCalculator = RateCalculator()
 ) {
 
-    fun calculate(tariff: DayBasedTariff, rentalStart: Instant, rentalEnd: Instant): Receipt {
+    fun calculate(tariff: DayBasedTariff, rentalPeriod: RentalPeriod): Receipt {
 
         val timeZoneId = tariff.timeZone.toZoneId()
 
-        val startLocalInstant = rentalStart.atZone(timeZoneId)
-        val endLocalInstant = rentalEnd.atZone(timeZoneId)
+        val startLocalInstant = rentalPeriod.calculatedStart.atZone(timeZoneId)
```

Comment:
Can we think of better names for this?

Reply:
I have revised the entire DayBasedTariffCalculator. Should I merge it in right away or should we finish this request first?

Reply:
It would be easier to keep this PR small. So let's review the changes after the merge. But we can resolve the issues for this class here so we will have a detailed look at this class in the next PR 🔍 , ok?